NewForce-at-Mountwest/bangazon-api-radvioli
# Ds bang

# Description

Created Customer controller, which allows users to create, edit, delete, customers. Controller also allows users to get all customers, get a single customer( by Id, or with query string by firstName or lastName), and allows for inclusion of Products or Payments associated with the Customer.


## Related Ticket 1: Allow developers to access the customer resource


## Steps to Test
git add and git commit what you're working on on your branch
git checkout master to switch to your local master branch
git fetch --all will give you access to your teammate's branches (if they've pushed them up)
git checkout ds-bang to access my branch
If you don't see the most recent version of their files, try git pull origin ds-bang
Run your server on their branch and make sure everything looks okay
Once server is running, use your browser to navigate to api/customers to ensure that correct information is being displayed
to add payments to customer, append this to end of search string: ?_include=products

to add payments to customer, append this to end of search string:  ?_include=payments

to query customer by pattern matching partial firstName or lastName append this to end of search string, followed by name fragment: ?q=




# Checklist:
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] My changes generate no new errors
- [x ] I have added tests that prove my fix is effective or that my feature works
- [x ] New and existing unit tests pass locally with my changes
